articles/introduction-to-evolutionary-algorithms/index.md

```diff
@@ -0,0 +1,404 @@
+while generations < 1000:
+    f_scores = [fitness(val, target) for val in population]
+    pool = selection(population)
+    generations = generation(pool, generations)
+    bestVal, bestScore = evaluate()
+    print("Gen: ", generations, population[bestVal], bestScore)
+    if bestS < bestScore:
+        bestV = population[bestVal]
+        bestS = bestScore
+    if bestScore == len(target):
+        break
+print("\n Best Match: ", bestV, bestS)
+print("The time now: ", time.asctime(time.localtime(time.time())))
+print("Total time taken: ", time.time() - initTime)
+
+````
+The output:
+````
+The time now:  Mon May 10 09:32:25 2021
+The population:
+['tlRWu', 'kBDkM', 'YxePb', 'TyYsM', 'UpmBu', 'yInSD', 'rOfLM', 'yLovB', 'vtMBA', 'LhVUL', 'TPFQv', 'xrBOZ', 'qHMdC', 'YSHlV', 'uyOLR', 'rTczE', 'aokjT', 'QvJvS', 'vffVO', 'eSnFy', 'IaBmO', 'zXkzL', 'IlYzI', 'DMEKO', 'sSFGW', 'aWlAm', 'QJFTH', 'ruKJl', 'SfGPJ', 'huOOi', 'OfNdu', 'UpqKP', 'JIYcg', 'UJorD', 'MJZro', 'NGGcU', 'MQQKs', 'twSRo', 'EueKl', 'umvDz', 'iDNrM', 'MjbON', 'Hzyht', 'Onrls', 'bUgSl', 'bhZPk', 'fnSXd', 'xvgmz', 'AvjNj', 'etFMZ']
+ Generations:
+Gen:  1 HzZro 2
+Gen:  2 Hzylo 3
+Gen:  3 Hzrlo 3
+Gen:  4 HJrlo 3
+Gen:  5 HAHlo 3
+Gen:  6 Hzylo 3
+Gen:  7 HeHlo 4
+Gen:  8 HeHlo 4
+Gen:  9 HeUlo 4
+Gen:  10 HeDlo 4
+Gen:  11 HeHlo 4
+Gen:  12 HeUlo 4
+Gen:  13 HeUlo 4
+Gen:  14 HeHlo 4
+Gen:  15 HeBlo 4
+Gen:  16 HePlo 4
+Gen:  17 Heplo 4
+Gen:  18 Hemlo 4
+Gen:  19 HeKlo 4
+Gen:  20 HeHlo 4
+Gen:  21 HeYlo 4
+Gen:  22 HeRlo 4
+Gen:  23 HeYlo 4
+Gen:  24 Hello 5
+ Best Match:  Hello 5
+The time now:  Mon May 10 09:32:25 2021
+Total time taken:  0.0304110050201416
+````
+You can see that the algorithm took only 0.03 s to compute the output.
+We have successfully dissected and coded an EA. Our monkey finally learned to say 'Hello'.
+Yay! 🥳
+
+Oh, and just in case you were wondering the real life effects of this thought experiment. In 2002, the University of Plymouth MediaLab Arts course studied the literary output of real monkeys. Well, monkeys are monkeys after all. Not only did they produce 5 pages of the letter S, they started pounding the keyboard with a stone. The director of the university's Institute of Digital Arts and Technology (i-DAT), said that monkeys
+> "are not random generators. They're more complex than that. ... They were quite interested in the screen, and they saw that when they typed a letter, something happened. There was a level of intention there."
+
+You can have look at the monkey's work [here](https://web.archive.org/web/20090318143423/http://www.vivaria.net/experiments/notes/publication/NOTES_EN.pdf)
+
+### Some fascinating outcomes of EA
+Evolutionary Algorithms are basically a means of asking the computer to think.
+Sometimes these have given rise to some totally out-of-the-box solutions.
+Have a look at some of these. These made me gape in awe!
+
+![walk-without-legs](/engineering-education/introduction-to-evolutionary-algorithms/EA.gif)
+
```

Comment:
Are all the images yours? If, not, make sure to add their sources

Reply:
Most of them are mine. I will add the sources for the others